crates/holochain/tests/tests/init.rs

```diff
@@ -67,3 +103,187 @@ async fn call_init_from_init_across_cells() {
 
     assert_eq!(inits.load(Ordering::SeqCst), 2);
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn call_init_with_invalid_return_type() {
+    let config = SweetConductorConfig::standard().no_dpki();
+    let mut conductor = SweetConductor::from_config(config).await;
+    let agent = SweetAgents::one(conductor.keystore()).await;
+    let zome = SweetInlineZomes::new(vec![], 0)
+        .function("init", |_, _: ()| Ok(42))
+        .function("touch", |_, _: ()| {
+            // Simple Zome to just trigger a call to init.
+            Ok(())
+        });
+
+    let (dna, _, _) = SweetDnaFile::unique_from_inline_zomes(zome).await;
+    let app = conductor
+        .setup_app_for_agent("app", agent, &[dna])
+        .await
+        .unwrap();
+    let (cell,) = app.into_tuple();
+
+    let err = conductor
+        .call_fallible::<_, ()>(&cell.zome(SweetInlineZomes::COORDINATOR), "touch", ())
+        .await
+        .unwrap_err();
+
+    let_assert!(ConductorApiError::CellError(CellError::WorkflowError(workflow_err)) = err);
+    let_assert!(
+        WorkflowError::RibosomeError(RibosomeError::CallbackInvalidReturnType(err_msg)) =
+            *workflow_err
+    );
+    assert!(err_msg == "invalid value: integer `42`, expected variant index 0 <= i < 3");
```

Comment:
That would be really nice but at the point where I create the `CallbackInvalidReturnType` error the only context we have is the zome name and this string from the deserialise error so I think I'd have to parse the string and get the necessary details from it. I'd hope we'd at least show the `CallbackInvalidReturnType` error so that will print `The callback has an invalid return type: invalid value: integer `42`, expected variant index 0 <= i < 3` which is a bit better.